extensions/default/src/DicomWebDataSource/utils/StaticWadoClient.ts

```diff
@@ -156,18 +156,44 @@ export default class StaticWadoClient extends api.DICOMwebClient {
    *
    * @param {*} desired
    * @param {*} actual
+   * @param {*} fuzzyMatching - if true, then do a sub-string match on the values
    * @returns true if the values match
    */
-  compareValues(desired, actual) {
+  compareValues(desired, actual, fuzzyMatching) {
     if (Array.isArray(desired)) {
-      return desired.find(item => this.compareValues(item, actual));
+      return desired.find(item => this.compareValues(item, actual, fuzzyMatching));
     }
     if (Array.isArray(actual)) {
-      return actual.find(actualItem => this.compareValues(desired, actualItem));
+      return actual.find(actualItem => this.compareValues(desired, actualItem, fuzzyMatching));
     }
     if (actual?.Alphabetic) {
       actual = actual.Alphabetic;
     }
+
+    if (fuzzyMatching && typeof actual === 'string' && typeof desired === 'string') {
+      const normalizeValue = str => {
+        if (str.startsWith('*')) {
+          str = str.slice(1);
+        }
+        if (str.endsWith('*')) {
+          str = str.slice(0, -1);
+        }
+        return str.toLowerCase();
+      };
+
+      const normalizedDesired = normalizeValue(desired);
+      const normalizedActual = normalizeValue(actual);
+
+      const tokenizeAndNormalize = str => str.replace(/\^/g, ' ').split(/\s+/).filter(Boolean);
+
+      const desiredTokens = tokenizeAndNormalize(normalizedDesired);
+      const actualTokens = tokenizeAndNormalize(normalizedActual);
+
+      return desiredTokens.some(desiredToken =>
```

Comment:
This should be desiredTokens.every - I know that the GCP definition has it with any, but that doesn't make any sense because it should be possible to provide a list of values to support "any", while it is important to allow multiple values to be done using every - think of patients whose first name is "John", and we are looking for "John Smith"
If it is done with John Smith as the query field, then some will find all the johns AND all the smiths, not the john smiths.